Automattic/wp-calypso
# media: migrate `updateMedia` to use React Query

This PR is part of a series of PRs to migrate the media library to React Query. I'm going to tackle mutations first and integrate them intermediately with state which currently still is in Redux. That way we'll get rid of all the Redux related code step by step similar to how we migrated the media library from Flux to Redux.

#### Changes proposed in this Pull Request

* media: migrate `updateMedia` to use React Query
* Remove Redux action creators, thunks and data layer parts related to `updateMedia`

#### Testing instructions

Please note that `updateMedia` only handles meta data updates like alt texts, captions, etc. It's not responsible to handle media edits (e.g. crop or change the size of an image).

* Go to `/media`, select one or more images and click _Edit_
* Changes to any of the detail fields should trigger a network request (same if you click on _Done_)
* Make sure those changes are persisted
* **Note**: you make changes from the editor modal blocks should get updated (best use _Caption_ to test)

Additionally, test the gallery edit when using a Classic block:

* Open the editor and add a Classic block
* If you add more than 1 image, you'll see the gallery view
* Hit _Edit_ and make changes to the _Caption_
* This should trigger a network request to update media data

Related to #58574
